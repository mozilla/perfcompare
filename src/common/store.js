import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

import search from '../reducers/searchSlice';

const reducer = combineReducers({
  search,
});

export default configureStore({
  reducer,
});
