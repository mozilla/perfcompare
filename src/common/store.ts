import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';

import comparison from '../reducers/ComparisonSlice';
import selectedRevisions from '../reducers/SelectedRevisionsSlice';
import theme from '../reducers/ThemeSlice';

const reducer = combineReducers({
  theme,
  selectedRevisions,
  comparison,
});

export const store = configureStore({
  reducer,
});

export const createStore = () => configureStore({ reducer });

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type Store = typeof store;
