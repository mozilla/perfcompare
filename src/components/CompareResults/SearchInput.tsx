import Close from '@mui/icons-material/Close';
import SearchIcon from '@mui/icons-material/Search';
import { FormControl, InputAdornment, TextField, Button } from '@mui/material';
import { style } from 'typestyle';

import { ButtonsLightRaw, Spacing } from '../../styles';

const styles = {
  container: style({
    display: 'flex',
    marginRight: 'auto',
    $nest: {
      '.form-container': {
        maxWidth: '447px',
      },
      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
      '.MuiButtonBase-root': {
        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
        ...ButtonsLightRaw.Secondary,
        marginLeft: Spacing.Small,
      },
    },
  }),
};

function SearchInput() {
  return (
    <div className={styles.container} data-testid={'search-by-title-test-name'}>
      <div className='form-container'>
        <FormControl variant='outlined'>
          <TextField
            disabled
            placeholder='Search by title'
            id='search-revision-input'
            size='small'
            InputProps={{
              startAdornment: (
                <InputAdornment position='start'>
                  <SearchIcon />
                </InputAdornment>
              ),
              endAdornment: (
                <InputAdornment position='end'>
                  <Close />
                </InputAdornment>
              ),
            }}
          />
        </FormControl>
      </div>
      <Button disabled>Apply</Button>
    </div>
  );
}

export default SearchInput;
