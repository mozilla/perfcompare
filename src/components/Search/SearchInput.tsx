import SearchIcon from '@mui/icons-material/Search';
import FormControl from '@mui/material/FormControl';
import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import { style } from 'typestyle';

import { useAppSelector } from '../../hooks/app';
import useHandleChangeSearch from '../../hooks/useHandleChangeSearch';
import { InputStylesRaw } from '../../styles';
import { InputType, ThemeMode, View } from '../../types/state';

interface SearchInputProps {
  onFocus: () => unknown;
  inputPlaceholder: string;
  view: View;
  mode: ThemeMode;
  searchType: InputType;
}

function SearchInput({
  onFocus,
  view,
  mode,
  inputPlaceholder,
  searchType,
}: SearchInputProps) {
  const { handleChangeSearch } = useHandleChangeSearch();
  const searchState = useAppSelector((state) => state.search[searchType]);
  const { inputError, inputHelperText, repository } = searchState;

  const size = view == 'compare-results' ? 'small' : undefined;

  const styles = {
    container: style({
      $nest: {
        '.hide': {
          visibility: 'hidden',
        },
        '.search-text-field': {
          width: '100%',
        },
        '.MuiInputBase-root': {
          ...(mode == 'light' ? InputStylesRaw.Light : InputStylesRaw.Dark),
          flexDirection: 'row',
        },
      },
    }),
  };

  return (
    <FormControl className={styles.container} fullWidth>
      <TextField
        error={inputError}
        helperText={inputError && inputHelperText}
        placeholder={inputPlaceholder}
        id={`search-${searchType}-input`}
        onFocus={onFocus}
        onChange={(e) => handleChangeSearch({ e, searchType, repository })}
        size={size}
        name={`${searchType}Search`}
        className={`search-text-field ${searchType}`}
        InputProps={{
          startAdornment: (
            <InputAdornment position='end'>
              <SearchIcon />
            </InputAdornment>
          ),
        }}
      />
    </FormControl>
  );
}

export default SearchInput;
