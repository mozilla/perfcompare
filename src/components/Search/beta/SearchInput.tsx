import type { Dispatch, SetStateAction } from 'react';

import SearchIcon from '@mui/icons-material/Search';
import FormControl from '@mui/material/FormControl';
import InputAdornment from '@mui/material/InputAdornment';
import TextField from '@mui/material/TextField';
import { connect } from 'react-redux';
import { style } from 'typestyle';

import type { RootState } from '../../../common/store';
import useHandleChangeSearch from '../../../hooks/useHandleChangeSearch';
import { Strings } from '../../../resources/Strings';
import { InputStylesRaw, Spacing } from '../../../styles';

const strings = Strings.components.searchDefault.base.collapedBase;

function SearchInput(props: SearchInputProps) {
  const { setFocused, inputError, inputHelperText, view, mode } = props;
  const { handleChangeSearch } = useHandleChangeSearch();
  const size = view == 'compare-results' ? 'small' : undefined;

  const styles = {
    container: style({
      $nest: {
        '.hide': {
          visibility: 'hidden',
        },
        '.search-text-field': {
          width: '100%',
          marginTop: `${Spacing.xSmall / 2}px`,
        },
        '.MuiInputBase-root': {
          ...(mode == 'light' ? InputStylesRaw.Light : InputStylesRaw.Dark),
          flexDirection: 'row',
        },
      },
    }),
  };
  return (
    <FormControl className={styles.container} fullWidth>
      <div className='hide'>Block</div>
      <TextField
        error={inputError}
        helperText={inputHelperText}
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
        placeholder={strings.inputPlaceholder}
        id='search-revision-input'
        onFocus={() => setFocused(true)}
        onChange={(e) => handleChangeSearch(e)}
        size={size}
        className='search-text-field'
        InputProps={{
          startAdornment: (
            <InputAdornment position='end'>
              <SearchIcon />
            </InputAdornment>
          ),
        }}
      />
    </FormControl>
  );
}

interface SearchInputProps {
  setFocused: Dispatch<SetStateAction<boolean>>;
  inputError: boolean;
  inputHelperText: string;
  view: 'compare-results' | 'search';
  mode: 'light' | 'dark';
}

function mapStateToProps(state: RootState) {
  return {
    inputError: state.search.inputError,
    inputHelperText: state.search.inputHelperText,
    searchResults: state.search.searchResults,
  };
}

export default connect(mapStateToProps)(SearchInput);
