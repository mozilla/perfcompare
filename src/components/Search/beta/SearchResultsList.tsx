import Box from '@mui/material/Box';
import List from '@mui/material/List';

import { SelectListLight, SelectListDark } from '../../../styles';
import type { Revision } from '../../../types/state';
import SearchResultsListItem from './SearchResultsListItem';

function SearchResultsList(props: SearchResultsListProps) {
  const { searchResults, view, mode, base } = props;

  return (
    <Box
      className={mode == 'light' ? SelectListLight : SelectListDark}
      id='search-results-list'
      alignItems='flex-end'
    >
      <List dense={view == 'compare-results'} sx={{ paddingTop: '0' }}>
        {searchResults.map((item, index) => (
          <SearchResultsListItem
            key={item.id}
            index={index}
            item={item}
            view={view}
            base={base}
          />
        ))}
      </List>
    </Box>
  );
}

interface SearchResultsListProps {
  searchResults: Revision[];
  view: 'compare-results' | 'search';
  mode: 'light' | 'dark';
  base: 'base' | 'new';
}

export default SearchResultsList;
