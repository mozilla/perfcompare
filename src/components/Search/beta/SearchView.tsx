import PerfCompareHeader from '../../Shared/beta/PerfCompareHeader';

function SearchViewBeta() {
  return (
    <section className='perfcompare-body'>
      <PerfCompareHeader />
    </section>
  );
}

export default SearchViewBeta;
