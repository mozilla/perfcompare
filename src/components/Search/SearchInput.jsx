import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import Container from '@mui/material/Container';
import InputLabel from '@mui/material/InputLabel';
import FormControl from '@mui/material/FormControl';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';
import TextField from '@mui/material/TextField';

import SearchResultsList from './SearchResultsList';

import SearchViewHelper from '../../utils/SearchViewHelper';

const { handleChangeDropdown, handleChangeSearch } = SearchViewHelper;

function SearchInput(props) {
  const { repository } = props;
  const repoList = ['try', 'autoland', 'mozilla-central'];

  return (
    <Container maxWidth="md">
      <FormControl sx={{ width: '25%' }}>
        <InputLabel id="select-repository">repository</InputLabel>
        <Select
          value={repository}
          labelId="select-repository"
          label="repository"
        >
          {repoList.length > 0 &&
            repoList.map((item) => (
              <MenuItem
                id={item}
                value={item}
                key={item}
                onClick={handleChangeDropdown}
              >
                {item}
              </MenuItem>
            ))}
        </Select>
      </FormControl>
      <TextField
        label="Search By Revision ID or Author Email"
        variant="outlined"
        sx={{ width: '75%' }}
        onChange={handleChangeSearch}
      />
      <SearchResultsList />
    </Container>
  );
}

SearchInput.propTypes = {
  repository: PropTypes.string.isRequired,
};

function mapStateToProps(state) {
  return {
    repository: state.search.repository,
  };
}

export default connect(mapStateToProps)(SearchInput);
