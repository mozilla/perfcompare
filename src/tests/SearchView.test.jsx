/**
 * @jest-environment jsdom
 */

import React from 'react';
import userEvent from '@testing-library/user-event';

import { render, fireEvent, screen } from './utils/test-utils';
import store from '../common/store';

import SearchView from '../components/Search/SearchView';
import SearchInput from '../components/Search/SearchInput';
import SearchResultsList from '../components/Search/SearchResultsList';
import {
  updateRepository,
  updateSearchResults,
  updateSearchValue,
} from '../reducers/searchSlice';

const unmockedFetch = global.fetch;

beforeEach(() => {
  global.fetch = () =>
    Promise.resolve(
      Promise.resolve({
        json: () => [],
      }),
    );
});

afterEach(() => {
  global.fetch = unmockedFetch;
  store.dispatch(updateSearchValue(''));
  store.dispatch(updateSearchResults([]));
  store.dispatch(updateRepository(''));
  jest.clearAllMocks();
  jest.resetAllMocks();
  jest.restoreAllMocks();
});

describe('Search View', () => {
  it('renders correctly', () => {
    const searchView = render(<SearchView />);

    // Title appears
    expect(screen.getByText(/PerfCompare/i)).toBeInTheDocument();

    // Repository Select appears
    expect(screen.getByLabelText(/Repository/i)).toBeInTheDocument();

    // No repositories are selected and dropdown is not visible
    expect(screen.queryByText(/autoland/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/try/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/mozilla-central/i)).not.toBeInTheDocument();
    // Search input appears
    expect(
      screen.getByLabelText(/Search By Revision ID or Author Email/i),
    ).toBeInTheDocument();

    // No list items should appear
    expect(screen.queryByRole('listitem')).not.toBeInTheDocument();

    expect(searchView).toMatchSnapshot();
  });
});

describe('Search Input', () => {
  it('should fetch recent revisions when a repository is selected', async () => {
    const spyOnFetch = jest.spyOn(global, 'fetch').mockImplementation(() =>
      Promise.resolve(
        Promise.resolve({
          json: () => ({
            results: [],
          }),
        }),
      ),
    );
    const user = userEvent.setup();
    render(<SearchInput />);
    const spyOnDispatch = jest.spyOn(store, 'dispatch');
    const spyOnGetState = jest.spyOn(store, 'getState');

    // Click Dropdown
    fireEvent.mouseDown(screen.getByRole('button', { name: 'repository' }));

    // Menu items should be visible
    expect(screen.getByText(/autoland/i)).toBeInTheDocument();
    expect(screen.getByText(/try/i)).toBeInTheDocument();
    expect(screen.getByText(/mozilla-central/i)).toBeInTheDocument();

    // Should call fetch, getState, and dispatch when an item is clicked
    await user.click(screen.getByRole('option', { name: 'autoland' }));
    expect(spyOnFetch).toHaveBeenCalledTimes(1);
    expect(spyOnGetState).toHaveBeenCalled();
    expect(spyOnDispatch).toHaveBeenCalledTimes(2);
  });
});

describe('Search Results List', () => {
  it('should show results if searchResults is not empty', () => {
    const results = [
      {
        id: 1,
        revision: 'coconut',
        author: 'johncleese@python.com',
        push_timestamp: 42,
        repository_id: 4,
      },
      {
        id: 2,
        revision: 'spam',
        author: 'ericidle@python.com',
        push_timestamp: 42,
        repository_id: 3,
      },
    ];
    store.dispatch(updateSearchResults(results));

    const searchResultsList = render(<SearchResultsList store={store} />);

    expect(store.getState().search.searchResults).toBe(results);
    expect(screen.getByText('coconut')).toBeInTheDocument();
    expect(screen.getByText('spam')).toBeInTheDocument();

    expect(searchResultsList).toMatchSnapshot();
  });
});
