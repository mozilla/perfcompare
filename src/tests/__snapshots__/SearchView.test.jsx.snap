// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results List should show results if searchResults is not empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style=""
  >
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
        />
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-l1oafp"
          >
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            >
              <div
                class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
                >
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      coconut
                    </span>
                  </div>
                </li>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
              <div
                class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
                role="button"
                tabindex="0"
              >
                <li
                  class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
                >
                  <div
                    class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                    >
                      spam
                    </span>
                  </div>
                </li>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-1equabv-MuiGrid-root"
      />
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 css-1qpzhgb-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-l1oafp"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    coconut
                  </span>
                </div>
              </li>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
            <div
              class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-cvhtoe-MuiButtonBase-root-MuiListItemButton-root"
              role="button"
              tabindex="0"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    spam
                  </span>
                </div>
              </li>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Search View renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiBox-root css-0"
        >
          <div
            class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter MuiTypography-gutterBottom css-1bn4tuo-MuiTypography-root"
          >
            PerfCompare
          </div>
        </div>
        <div
          class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
        >
          <div
            class="MuiFormControl-root css-x1sfaz-MuiFormControl-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              id="select-repository"
            >
              repository
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="select-repository"
                class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                role="button"
                tabindex="0"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    repository
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-160oy9s-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="mui-1"
              id="mui-1-label"
            >
              Search By Revision ID or Author Email
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1ftyaf0"
                >
                  <span>
                    Search By Revision ID or Author Email
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiTypography-root MuiTypography-h1 MuiTypography-alignCenter MuiTypography-gutterBottom css-1bn4tuo-MuiTypography-root"
        >
          PerfCompare
        </div>
      </div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
      >
        <div
          class="MuiFormControl-root css-x1sfaz-MuiFormControl-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            id="select-repository"
          >
            repository
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="select-repository"
              class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  repository
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-160oy9s-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="mui-1"
            id="mui-1-label"
          >
            Search By Revision ID or Author Email
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-1"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Search By Revision ID or Author Email
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
